new changes for B3 WALDER project to make changes for discovery porta…

urvi-occ opened the pull request:
Integrated the Discovery Portal into the Chicagoland Pandemic Response Commons (PRC) under the B3 project, enabling advanced search functionality with "AND" and "OR" operations.

- **New Button Added into Discovery Portal Action Bar:**
Introduced the `OpenFillRequestFormButton` in the Discovery action bar component, allowing users to submit a fill request form with dynamic query generation based on selected resources. The button is conditionally rendered based on configuration settings.

- Introduced the following variables in the config
    - **enableFillRequestForm**: This enables the button to open a form for requesting access to a resource for PRC.
    - **fillRequestFormDisplayText**: Text displayed on the button for requesting information.
    - **fillRequestFormCheckField**: MDS field used to validate whether entries in the Discovery Portal have unique IDs.
    - **fillRequestFormURL**: URL of the form used for requesting access.
    - **externalWebsiteName**: Name of an external website shown in a popover when the form is enabled.
    - **externalWebsiteURL**: URL of the external website displayed with its name when the form is enabled.
  
### Improvements
- Introduced the `OpenFillRequestFormButton` in the Discovery action bar component, allowing users to submit a fill request form with dynamic query generation based on selected resources.
- Add a few config fields that are related to `OpenFillRequestFormButton`. See `docs/portal_config.md` for details

urvi-occ commented:
OCC is ready to test this in PRC QA, can I get an automated copy or anything so this PR can pass the test and I can use it in QA for testing? I am looking for something similar on how it was used for PR #1392 and #1394 which can be seen [here](https://github.com/uc-cdis/gitops-qa/compare/master...urvi-occ:gitops-qa:feat/b3_walder_discovery_prc_qa_covid19?expand=1)

cc @atharvar28 @paulineribeyre

paulineribeyre commented:
Created temp branch for testing: `automatedCopy-feat/b3_walder_discovery_prc_2`

mfshao commented on src/Discovery/DiscoveryActionBar/components/OpenFillRequestFormButton.tsx:
```diff
@@ -0,0 +1,52 @@
+const OpenFillRequestFormButton = ({
+    props,
+  }) => (
+  props.config.features.exportToWorkspace?.openFillRequestForm && (
```
```suggestion
  props.config.features.exportToWorkspace?.enableFillRequestForm && (
```

urvi-occ replied:
Change done, I will commit together once all of this is resolved

mfshao commented on src/Discovery/DiscoveryActionBar/components/OpenFillRequestFormButton.tsx:
```diff
@@ -0,0 +1,52 @@
+        <React.Fragment>
+          &nbsp;
+          <a target='_blank' rel='noreferrer' href='https://pandemicresponsecommons.org'>
+            {'Pandemic Response Commons Website'}
```
I hope we don't have to hardcode these env specific values in here
the url can be inferred from localconf, the text might be also about to be pulled from some value there
if not, maybe put them in configs

urvi-occ replied:
I was planning to put it in config but I wasnt sure that it would be a good idea to just have config for value which will not change ever. I can add it into config if needed

urvi-occ replied:
Put variable in the config, add external link icon

mfshao replied:
external link icon in portal https://github.com/uc-cdis/data-portal/blob/master/src/components/layout/FooterNIAID.jsx#L162-L164

urvi-occ replied:
external link icon and config variable for website and its name are added

mfshao commented on src/Discovery/DiscoveryActionBar/components/OpenFillRequestFormButton.tsx:
```diff
@@ -0,0 +1,52 @@
+              combinedIds += ',';
+            }
+            combinedIds += item._medical_sample_id;
+          });
```
L29-35 can be replaced by
```
const combinedIds = props.discovery.selectedResources.map((item)=>(item._medical_sample_id)).join(',');
```

urvi-occ replied:
Change done, I will commit together once all of this is resolved

mfshao commented on src/Discovery/DiscoveryActionBar/components/OpenFillRequestFormButton.tsx:
```diff
@@ -0,0 +1,52 @@
+            }
+            combinedIds += item._medical_sample_id;
+          });
+          const url = `${props.config.features.exportToWorkspace.fillRequestFormURL}?query=${encodeURIComponent(combinedIds)}`;
```
```suggestion
          const url = `${props.config.features.exportToWorkspace.fillRequestFormURL}query=${encodeURIComponent(combinedIds)}`;
```
No need to check here, already done in L28 by the ternary operator

mfshao replied:
also `combinedIds` here might also be empty here, unless `_medical_sample_id` is guarantee to exists in the metadata of all selectable studies, consider if you need some extra error handling for that case here

urvi-occ replied:
That is not a ternary operator, that is a URL generated with a query parameter that requires a question mark

urvi-occ replied:
> also `combinedIds` here might be empty, unless `_medical_sample_id` is guaranteed to exists in the metadata of all selectable studies, consider if you need some extra error handling for that case here

OCC is pushing data into this mds by a semi-automated system that parses the file, appends a medical ID, and then pushes it to mds. Even though we want to put an error handling, I have to put a UI error handling in this case, can you tell me what reference in gen3 I can take and replicate here?

urvi-occ replied:
> No need to check here, already done in L28 by the ternary operator

That is not a ternary operator, that is a URL generated with a query parameter that requires a question mark

urvi-occ replied:
Disable checkbox if `_medical_sample_id` doesn't exist also add check into code if `combinedIds` are empty

urvi-occ replied:
the code already exists for that in discovery list view in [here](https://github.com/uc-cdis/data-portal/blob/5f3dfe2e2f15a0b7527f8d6dcbc10b6fa4ec6d98/src/Discovery/DiscoveryListView.tsx#L77-L83)

mfshao replied:
the code you have in the link will only enable/disable checkboxs based on the existance of the file manifest fields. It has nothing to do with the `_medical_sample_id` field which you really cares in this feature

urvi-occ replied:
so the `manifestFieldName` here is going to be the `_medical_sample_id` and its doing the checks, you can inspect the code on [https://qa-covid19.planx-pla.net/discovery](https://qa-covid19.planx-pla.net/discovery) and check. I would have appended the video of it but the discovery page is down for some reason

mfshao replied:
And I think QA is down because of this https://github.com/uc-cdis/data-portal/pull/1551#discussion_r1824760589

urvi-occ replied:
So for your question about `manifestField` Name, it is defined in the gitops config [here](https://github.com/uc-cdis/gitops-qa/blob/b66688c0f8200cd4bb4d9c7f8b53548a7f9ad23d/qa-covid19.planx-pla.net/portal/gitops.json#L1324) I have referred other manifest files from bloodpac and heal and this is how its configured in those as well. As DiscoveryListView file is used to generate list view in different commons, I used the same concept. I am getting really confused with multiple comments here on different lines here.

mfshao replied:
What I'm trying to say is that the field in metadata that has this name of `manifestField` should have its value as Gen3 file manifests. For example, we usually have `"manifestField": "__manifest"` in portal config and the value of `"__manifest"` in metadata is a Gen3 file manifest (which has some predefined field like `title`, `description`, `object_id`, etc.).
In your case, I don't think the value of `"_medical_sample_id"` in your metadata will be a Gen3 file manifest right? I might be wrong since I haven't seen your metadata in MDS, but the field name in metadata doesn't sound like a manifest.
If the field in metadata does not represent a manifest, don't call it `manifestField`
But if the field of `"_medical_sample_id"` in your metadata did represents a file manifest, then what I have said is not right. But you might want to consider renaming your metadata field to make sure its name indicates its value is a file manifest.

urvi-occ replied:
@mfshao 

This is my record in mds in PRC QA: [here](https://qa-covid19.planx-pla.net/mds/metadata/0001-10218-buffycoat)

When I created this mds page, I took reference from both BLOODPAC and HEAL.

In BLOODPAC if you see [here](https://github.com/urvi-occ/cdis-manifest/blob/3eaa7480eedd9e2bebd7b9be313c24bdcd546324/data.bloodpac.org/portal/gitops.json#L1074) , They have used field `_publication_id` as an aggregation field. 

I am using both of these as reference as there are no particular standard for this. This should have been reviewed when I was pushing this to QA. Can we please move this for now and I can make changes later as I have a deadline.

urvi-occ replied:
I don't have any files, so I cant use the same format in `__manifest`, can you suggest from [this](https://qa-covid19.planx-pla.net/mds/metadata/0001-10218-buffycoat) what can be the exact format for `__manifest`

urvi-occ replied:
I made a PR and added `__manifest` in [this](https://qa-covid19.planx-pla.net/mds/metadata/0001-10218-buffycoat) the [PR](https://github.com/uc-cdis/gitops-qa/pull/3060) can you please approve it?

mfshao replied:
I can approve the gitops-qa PR but I think it is unnecessary for your case, and updating your PRC metadata by adding the `__manifest` field in it is also unnecessary.
I'll try to put this again, the correct way to address this issue is: in the logic of checking whether a checkbox should be disabled, DO NOT rely only one checking whether a `manifestField` exist or not in metadata as what https://github.com/uc-cdis/data-portal/blob/f0b0974ace27f7ace9d990be86cf50cbd863300b/src/Discovery/DiscoveryListView.tsx#L80-L90 is doing now, if the `OpenFillRequestFormButton` is enabled
If the `OpenFillRequestFormButton` button is enabled, you should be specifically checking if the field `_medical_sample_id` exists in metadata, in [here](https://github.com/uc-cdis/data-portal/blob/f0b0974ace27f7ace9d990be86cf50cbd863300b/src/Discovery/DiscoveryListView.tsx#L76-L79). Since this logic is only related to `OpenFillRequestFormButton`, it should be separated from the rest of the checkbox enable/disable check logic

urvi-occ replied:
and if the concern is about this code snippet below not applying to some other commons without that concerning other commons, if you can closely see it is using variable `enableFillRequestForm` which is configured and used as a part of this particular feature. I hope this is enough to satisfy the need here. 

https://github.com/uc-cdis/data-portal/blob/f0b0974ace27f7ace9d990be86cf50cbd863300b/src/Discovery/DiscoveryListView.tsx#L76-L79

mfshao replied:
> @mfshao I really think that I SHOULD NOT BE using, '_medical_sample_id`, a field from MDS inside a code as it would be a "static" value in the code and that would contradict your previous comments.

That is right. The most generic way to implement this feature is to put something like this `"fillRequestFormCheckField": "_medical_sample_id"` into portal config. And the lines at https://github.com/uc-cdis/data-portal/blob/f0b0974ace27f7ace9d990be86cf50cbd863300b/src/Discovery/DiscoveryListView.tsx#L76-L79 will use this `fillRequestFormCheckField` field to check whether a checkbox should be enabled or disabled when `enableFillRequestForm` is set. And that is consistant with what I was suggesting at the beginning.
But since then, the time you can spend on implenent this generic solution has shifted, now you are under a time pressure. And I think that at least for now, PRC will be the only one that needs this `OpenFillRequestFormButton`. so that is why I'm saying now I'm okay with hard coding the `_medical_sample_id` to use inside https://github.com/uc-cdis/data-portal/blob/f0b0974ace27f7ace9d990be86cf50cbd863300b/src/Discovery/DiscoveryListView.tsx#L76-L79. Plus, we are already hard coding `_medical_sample_id` in `OpenFillRequestFormButton.tsx` anyways.
So I'm trying to help you here by taking some complexity off from your plate. But if you prefer not to hard code `_medical_sample_id` and go with the more generic ways, then I'm also fine with it

mfshao replied:
Anyway, if we are really having a hard time to implement this. You can put a few `# TODO` or `# FIXME` comments in the proper files to remind us we need to get rid of hard coded `_medical_sample_id` field in the future, and patch up the checkbox enable/disable validation in `DiscoveryListView.tsx`. That way we can pick them up later and get this merged. I agree maybe this is good enough for PRC for now

urvi-occ replied:
I think I found a way, let me commit the changes soon.

mfshao commented on src/Discovery/DiscoveryActionBar/components/OpenFillRequestFormButton.tsx:
```diff
@@ -0,0 +1,52 @@
+          const url = `${props.config.features.exportToWorkspace.fillRequestFormURL}?query=${encodeURIComponent(combinedIds)}`;
+          window.open(url, '_blank');
+        } : () => {
+          window.open(props.config.features.exportToWorkspace.fillRequestFormURL, '_blank');
```
why it should open this URL in this case, where `fillRequestFormURL` doesn't exists?

urvi-occ replied:
Technically it should not redirect anywhere else as there is nothing setup

urvi-occ replied:
if `fillRequestFormURL` doesn't exist, hide the button request Access

urvi-occ replied:
I have added the code to hide button `fillRequestFormURL` [here](https://github.com/urvi-occ/data-portal/blob/56b658928439ed424a77b866523c3a49854e688b/src/Discovery/DiscoveryActionBar/components/OpenFillRequestFormButton.tsx#L10)

mfshao replied:
Even though you have put checks at the beginning of this function, I would still strongly suggest you should change this line of code. The falsy expression of this ternary operator should simply just be `() => {}`, the current way the code was written is very confusing

urvi-occ replied:
This is not a ternary operator it is a URL with parameters. I am really confused here on what this comment means

urvi-occ replied:
I have changed the code, which is now on L45 let me know if this make sense

mfshao replied:
honestly the simplest way to fix this is to remove L46 entriely, which mean if `fillRequestFormURL` doesn't exist, then this button does nothing

mfshao replied:
Fonctionality-wise this doen't change anything, since this button won't be rendered in the first place if `fillRequestFormURL` doesn't exist. So you could even consider remove the ternary operator from L41 entirely, which will make the code simpler and easier to understand

urvi-occ replied:
Can you please provide the exact code that you think will be a good here? I can review and add it in PR.

mfshao commented on src/Discovery/DiscoveryActionBar/components/OpenFillRequestFormButton.tsx:
```diff
@@ -0,0 +1,52 @@
+        }}
+        type='default'
+        className={`discovery-action-bar-button${(props.discovery.selectedResources.length === 0) ? '--disabled' : ''}`}
+        disabled={props.discovery.selectedResources.length === 0}
```
consider also disabling the button if `fillRequestFormURL` does not exists

urvi-occ replied:
I'll be replacing this

disabled={props.discovery.selectedResources.length === 0 || !props.config.features.exportToWorkspace.fillRequestFormURL }

mfshao commented on src/Discovery/DiscoveryActionBar/components/OpenFillRequestFormButton.tsx:
```diff
@@ -0,0 +1,52 @@
+        disabled={props.discovery.selectedResources.length === 0}
+        icon={<FileTextOutlined />}
+      >
+        {'Click Here to Access Request'}
```
```suggestion
        {'Click Here to Request Access'}
```
shouldn't it be put this way?

mfshao replied:
actually, this should be set dynamically by the value of `fillRequestFormDisplayText` right?

urvi-occ replied:
Change done, I will commit together once all of this is resolved

urvi-occ replied:
Change it to

mfshao commented on src/Discovery/DiscoveryActionBar/components/OpenFillRequestFormButton.tsx:
```diff
@@ -0,0 +1,52 @@
+        disabled={props.discovery.selectedResources.length === 0}
+        icon={<FileTextOutlined />}
+      >
+        {'Click Here to Access Request'}
```
`Request Access` is a very vague message. Better specify what the users need to request access for.

urvi-occ replied:
What would you suggest then?

mfshao commented on src/Discovery/DiscoveryListView.tsx:
```diff
@@ -47,8 +47,9 @@ const DiscoveryListView: React.FunctionComponent<Props> = (props: Props) => {
-              && props.config.features.exportToWorkspace.enabled
+        (props.config.features.exportToWorkspace
+          && props.config.features.exportToWorkspace.enabled) || ((props.config.features.exportToWorkspace?.enableFillRequestForm
+            && props.config.features.exportToWorkspace.enableFillRequestForm === true))
```
unnecessary double parentheses here and the line above

urvi-occ replied:
Change done, I will commit together once all of this is resolved

mfshao commented on src/Discovery/DiscoveryListView.tsx:
```diff
@@ -47,8 +47,9 @@ const DiscoveryListView: React.FunctionComponent<Props> = (props: Props) => {
-        props.config.features.exportToWorkspace
-              && props.config.features.exportToWorkspace.enabled
+        (props.config.features.exportToWorkspace
+          && props.config.features.exportToWorkspace.enabled) || ((props.config.features.exportToWorkspace?.enableFillRequestForm
```
we probably should consider renaming the config section `exportToWorkspace` to something else since it is more like an action button bar rather than just for exporting to ws
but this is more like a note to ourselves, not a item that in the scope of your PR

urvi-occ replied:
Are you still planning to merge PR when all comments are resolved? We’d recommend that as we have a deadline to be met

mfshao commented on docs/portal_config.md:
```diff
@@ -464,6 +464,9 @@ Below is an example, with inline comments describing what each JSON block config
           "enableDownloadManifest": true, // enables a button which allows user to download a manifest file for gen3 client
           "downloadManifestButtonText": "Download Manifest", // text to be displayed on the download manifest button
+          "enableFillRequestForm" : true, // enables a button which opens a new form to request access to a resoruce
+          "openFillRequestForm": "Request Access", // text to be displayed on fill the request form
```
```suggestion
          "fillRequestFormDisplayText": "Request Access", // text to be displayed on fill the request form
```

urvi-occ replied:
Change done, I will commit together once all of this is resolved

mfshao commented on src/Discovery/DiscoveryActionBar/components/OpenFillRequestFormButton.tsx:
```diff
@@ -0,0 +1,57 @@
+        </Button>
+      </Popover>
+    )
+  );
```
you need to explicitly return `null` if you want this component to render nothing when the conditions at the beginning doesn't meet

urvi-occ replied:
Added

mfshao commented on src/Discovery/DiscoveryActionBar/components/OpenFillRequestFormButton.tsx:
```diff
@@ -0,0 +1,57 @@
+          disabled={props.discovery.selectedResources.length === 0 || !props.config.features.exportToWorkspace.fillRequestFormURL || props.discovery.selectedResources.map((item) => (item._medical_sample_id)).join(',').length == 0}
+          icon={<FileTextOutlined />}
+        >
+          {'Click Here to ' + (props.config.features.exportToWorkspace.fillRequestFormDisplayText || 'Request Information')}
```
```suggestion
          {`Click Here to ${props.config.features.exportToWorkspace.fillRequestFormDisplayText || 'Request Information'}`)}
```
better user template strings

urvi-occ replied:
make sense. done

mfshao commented on src/Discovery/DiscoveryActionBar/components/OpenFillRequestFormButton.tsx:
```diff
@@ -0,0 +1,57 @@
+import React from 'react';
```
this whole file need some linting, consider running eslint over this file

urvi-occ replied:
Done

mfshao commented on src/Discovery/DiscoveryActionBar/components/OpenFillRequestFormButton.tsx:
```diff
@@ -0,0 +1,57 @@
+            const url = `${props.config.features.exportToWorkspace.fillRequestFormURL}?query=${encodeURIComponent(combinedIds)}`;
+            window.open(url, '_blank');
+          } : () => {
+            window.open(props.config.features.exportToWorkspace.fillRequestFormURL, '_blank');
```
```suggestion
```
see https://github.com/uc-cdis/data-portal/pull/1551#discussion_r1824749568

urvi-occ replied:
changes made

mfshao commented on src/Discovery/DiscoveryConfig.d.ts:
```diff
@@ -15,6 +15,9 @@ export interface DiscoveryConfig {
             enableDownloadVariableMetadata?: boolean
+            enableFillRequestForm?: boolean
+            fillRequestFormDisplayText?: string
+            fillRequestFormURL?: string
```
missing `externalWebsiteName` and `externalWebsiteURL` in here?

urvi-occ replied:
added

mfshao commented on src/Discovery/DiscoveryListView.tsx:
```diff
@@ -71,6 +72,11 @@ const DiscoveryListView: React.FunctionComponent<Props> = (props: Props) => {
+          // if enableFillRequestForm is true, we allow users to check the checkbox without login
+          if (props.config.features.exportToWorkspace?.enableFillRequestForm
+            && props.config.features.exportToWorkspace.enableFillRequestForm === true) {
+            disabled = false;
```
related to comment https://github.com/uc-cdis/data-portal/pull/1551#discussion_r1824754029

urvi-occ replied:
I have changed `_medical_sample_id` to `__manifest` in [this metadata](https://qa-covid19.planx-pla.net/mds/metadata/0001-10218-buffycoat) [the](https://github.com/uc-cdis/gitops-qa/pull/3060) can you please approve it?

mfshao replied:
see this comment https://github.com/uc-cdis/data-portal/pull/1551/files#r1832012660

mfshao reviewed (commented):
This branch is failing on some CI checks, pls see details here https://github.com/uc-cdis/data-portal/actions/runs/11615275573/job/32345448092?pr=1551

mfshao commented on src/Discovery/DiscoveryActionBar/components/OpenFillRequestFormButton.tsx:
```diff
@@ -0,0 +1,63 @@
+        className={`discovery-action-bar-button${(props.discovery.selectedResources.length === 0) ? '--disabled' : ''}`}
+        disabled={
+          props.discovery.selectedResources.length === 0 ||
+          !props.config.features.exportToWorkspace.fillRequestFormURL ||
```
you don't need this line here now, because the button won't be rendered according to L9

urvi-occ replied:
can you make all the changes you think are good enough and add them to the commit? I can review it. I am no expert on this and am having hard time understanding this

urvi-occ replied:
removed the code. does this look okay?

mfshao commented:
FYI this PR is still failing on some mandatory checks

urvi-occ commented:
> I think we are getting close to the finish line, so please also update the PR description so it can properly reflect what the PR is adding/changing exactly in a few bulletin points. This is importatnt since they will be pull in to generate release notes automatically. Here is an example how the PR description should look like #1588

Hey Mingfei, I followed the format and I don't have any JIRA tickets, bug fixes, or new features in the JIRA description. that's the reason I have written improvements. I have made it more clearer now. does that make sense

mfshao commented:
> > I think we are getting close to the finish line, so please also update the PR description so it can properly reflect what the PR is adding/changing exactly in a few bulletin points. This is importatnt since they will be pull in to generate release notes automatically. Here is an example how the PR description should look like #1588
> 
> Hey Mingfei, I followed the format and I don't have any JIRA tickets, bug fixes, or new features in the JIRA description. that's the reason I have written improvements. I have made it more clearer now. does that make sense

Yeah JIRA ticket is optional.
I would remove bulletin point 1 since it is too vague, or fill a bit more details into it by saying what has changed
And I would add the name of the new button `OpenFillRequestFormButton` into 2

urvi-occ commented:
I have added information and I have also added code for `fillRequestFormCheckField` with [PR](https://github.com/uc-cdis/gitops-qa/pull/3060). Does this all make more sense?

mfshao reviewed (approved):
code wise LGTM, waiting for results from CI

mfshao commented on src/Discovery/DiscoveryListView.tsx:
```diff
@@ -71,6 +72,18 @@ const DiscoveryListView: React.FunctionComponent<Props> = (props: Props) => {
+          if (props.config.features.exportToWorkspace?.enableFillRequestForm) {
+            disabled = false;
+          }
+          // Check if the fillRequestFormCheckField is empty, and disable the checkbox if true
+          const fillRequestFormCheckField = props.config.features.exportToWorkspace?.fillRequestFormCheckField;
+          if (fillRequestFormCheckField) {
+            const fieldValue = record[fillRequestFormCheckField];
+            if (!fieldValue || fieldValue.length === 0) {
+              disabled = true;
+            }
+          }
```
It would be better if this logic is put into the `if` block above

urvi-occ replied:
should I merge it with this

https://github.com/urvi-occ/data-portal/blob/df82da937789dd1215e3e403dc7ded8e1fd9a972/src/Discovery/DiscoveryListView.tsx#L76-L78


or this

https://github.com/urvi-occ/data-portal/blob/df82da937789dd1215e3e403dc7ded8e1fd9a972/src/Discovery/DiscoveryListView.tsx#L88-L96

mfshao replied:
into L76-78 should be more appropriate IMO

urvi-occ replied:
Done

mfshao commented on src/Discovery/DiscoveryListView.tsx:
```diff
@@ -72,19 +72,16 @@ const DiscoveryListView: React.FunctionComponent<Props> = (props: Props) => {
+          const enableFillRequestForm = props.config.features.exportToWorkspace?.enableFillRequestForm;
+
+          if (!enableFillRequestForm || (!record[fillRequestFormCheckField]?.length)) {
+            disabled = true;
```
I don't think this is right. `disabled` is defaulted to a falsy value, so you need to very careful when setting it explicitly to true
I believe the way you write currently will disable all the checkboxes if the `enableFillRequestForm` is not set in config

urvi-occ replied:
I did. please check now

urvi-occ replied:
I see what you are trying to say, this can create confusion, I have added a new property variable `enableCheckboxWithoutLogin` which will allow the discovery portal to enable checkboxes without logging in. is this a better solution?

mfshao replied:
No you don't need this extra field in config. What you have before is already doing this because of https://github.com/urvi-occ/data-portal/blob/df82da937789dd1215e3e403dc7ded8e1fd9a972/src/Discovery/DiscoveryListView.tsx#L76-L78
Sinply copy and paste this blcok https://github.com/urvi-occ/data-portal/blob/df82da937789dd1215e3e403dc7ded8e1fd9a972/src/Discovery/DiscoveryListView.tsx#L79-L86 into the if https://github.com/urvi-occ/data-portal/blob/df82da937789dd1215e3e403dc7ded8e1fd9a972/src/Discovery/DiscoveryListView.tsx#L76-L78 would do it, or just revert back to this commit https://github.com/uc-cdis/data-portal/pull/1551/commits/df82da937789dd1215e3e403dc7ded8e1fd9a972

urvi-occ replied:
done

mfshao commented on src/Discovery/DiscoveryActionBar/components/OpenFillRequestFormButton.tsx:
```diff
@@ -0,0 +1,63 @@
+    >
+      <Button
+        onClick={() => {
+          const combinedIds = selectedResources.map((item) => item[props.props.config.features.exportToWorkspace.fillRequestFormCheckField]).join(',');
```
```suggestion
          const combinedIds = selectedResources.map((item) => item[config.features.exportToWorkspace.fillRequestFormCheckField]).join(',');
```
you already have that from https://github.com/uc-cdis/data-portal/blob/2b467498fc9b54da4e0ddaacbe951f74f088abf1/src/Discovery/DiscoveryActionBar/components/OpenFillRequestFormButton.tsx#L7

mfshao replied:
or actually, get `fillRequestFormCheckField` from https://github.com/uc-cdis/data-portal/blob/2b467498fc9b54da4e0ddaacbe951f74f088abf1/src/Discovery/DiscoveryActionBar/components/OpenFillRequestFormButton.tsx#L20-L22

mfshao commented on src/Discovery/DiscoveryListView.tsx:
```diff
@@ -47,15 +47,22 @@ const DiscoveryListView: React.FunctionComponent<Props> = (props: Props) => {
         renderCell: (_checked, _record, _index, node) => (
           <Tooltip
-            title={`Click to select item for ${(props.config.features.exportToWorkspace.enableDownloadManifest || props.config.features.exportToWorkspace.enableDownloadZip) ? 'download or ' : ''}open in workspace`}
+            title={`Click to select item for ${
+              props.config.features.exportToWorkspace.enableFillRequestForm
+                ? props.config.features.exportToWorkspace.fillRequestFormDisplayText.toLowerCase()
+                : (props.config.features.exportToWorkspace.enableDownloadManifest || props.config.features.exportToWorkspace.enableDownloadZip)
+                  ? 'download or open in workspace'
+                  : 'open in workspace'
+              }`}
```
Actually reading this again, this is making to checkbox tooltip texts inaccurate when both the fill request form button and export/download button are enabled right?

urvi-occ replied:
Thanks for pointing that out! I have updated the tooltip to clearly list all enabled options when multiple actions (like "Fill Request Form" and "Download") are available.

mfshao commented on src/Discovery/DiscoveryListView.tsx:
```diff
@@ -47,15 +47,28 @@ const DiscoveryListView: React.FunctionComponent<Props> = (props: Props) => {
+            title={`Click to select item for ${
+              [
+                props.config.features.exportToWorkspace.enableFillRequestForm
+                  ? props.config.features.exportToWorkspace.fillRequestFormDisplayText.toLowerCase()
```
```suggestion
                  ? props.config.features.exportToWorkspace.fillRequestFormDisplayText?.toLowerCase()
```
`fillRequestFormDisplayText` can possibily be `undefined`

mfshao reviewed (approved):
lgtm 👍

mfshao commented:
CI passed in https://github.com/uc-cdis/data-portal/pull/1627
